README.md

```diff
@@ -1,15 +1,198 @@
-# Test corpus for tfhe-rs backward compatibility
-This repo holds various messages from tfhe-rs that have been versioned and serialized.
-The goal is to detect in tfhe-rs Ci when the version of a type should be upgraded because a breaking change has been added.
+# tfhe-rs backwards compatibility test corpus
+This repo contains various messages from [TFHE-rs](https://github.com/zama-ai/tfhe-rs) that have been versioned and serialized.
+The goal is to detect in TFHE-rs CI when the version of a type should be updated because a breaking change has been added.
+
+The messages are serialized using cbor and bincode because they both support large arrays and are vulnerable to different sets of breaking changes. Each message is stored with a set of metadata to verify that the values are loaded correctly.
+
+# Usage
+In TFHE-rs main repo, run the following command
+```
+make test_backward_compatibility
+```
+This will clone this repo and check if all messages are handled correctly. By default, this will use the `v0.1` branch (see below). To use a different branch, use this command instead:
+```
+BACKWARD_COMPAT_DATA_BRANCH=my_branch_name make test_backward_compatibility
+```
+
+# Versioning this repo
+The tests in this repo are by definition forward compatible (they should run on any future TFHE-rs release). They are also backward compatible (allowing tests to be run on past TFHE-rs versions, for example to bisect a bug), mostly because the TFHE-rs test driver will simply ignore any unknown test types and only load tests for versions inferior to its own.
+
+However, this does not allow changes to be made to the test metadata scheme itself. In such a case, a new version branch should be created (e.g. `v0.2`), and TFHE-rs should use that branch instead.
+
+Any commits to `main` should be backported to the latest version branch.
```

Comment:
what does that mean ?

Reply:
Since TFHE-rs uses the branch v0.1 we need to backport commits from main into v0.1, or they won't be used. If we add a breaking change to the metadata format we will have to create a v0.2 branch, and at that point we will have to backport commits into v0.2.

There might be an easier way to do this.

Reply:
well as long as main == v0.1 keep v0.1 on the same branch

the day it changes v0.1 stops moving and we put v0.2 at the tip of main, unless we expect to backport on v0.1 but once we switch to 0.2 it's a done deal right ?